client/src/pages/EnergiserDetail.js

```diff
@@ -14,6 +11,35 @@ import EnergizerDeleteButton from "./components/EnergizerDeleteButton";
 function EnergiserDetail() {
 	const { id } = useParams();
 	const [item, setItem] = useState(null);
+	const [showFeedback, setShowFeedback] = useState(false);
+	const [currentValue, setCurrentValue] = useState(0);
+	const [hoverValue, setHoverValue] = useState(undefined);
+	const stars = Array(5).fill(0);
+
+	const handleClick = (value) => {
+		setCurrentValue(value);
+		console.log(value);
+		fetch(`/api/energizers/${id}/rate`, {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({ value }),
+		})
+			.then((res) => res.json())
+			.then((data) => {
+				setItem(data);
```

Comment:
As per API comment below, I think you just need to update the average rating not the entire object.